implementation-guide/Implementation Guide.md

```diff
@@ -0,0 +1,632 @@
+
+Lets imagine that Alisha (Node A) also knows Carol (Node C). Carol has a gallery of interesting robot photos and is looking to grow so has asked Alisha if she can help. Since Ben (Node B) has some amazing robots, Alisha decides to ask Ben if he's able to send some photos of his robot over to Carol. Alisha is able to do this without needing to worry if Ben knows Carol, she can simply share a reference to Carol's robot gallery on Node C and Third Party Handoffs will ensure that Ben can secrely get the reference to the gallery.
+
+Third Party Handoffs has certain terms that are important to understand when thinking about handoffs, using the the above example, we'll look at who is who from the prospective of Alisha giving Ben a reference to Carol's gallery:
+
+- **Gifter**: This would be Alisha' Node, Node A it's the one initiating the handoff
+- **Receiver**: This would be Ben's Node, Node B as it's the one getting the reference
+- **Exporter**: This would be Carol's Node, Node C as it's the node the reference that's being shared lives on
+
+![](./handoffs-3party1.webm)
+
+In a high level Alisha will deposit a "gift" which is just a reference to the gallery object on Node C's gift table the session Node A and Node C have together. Alisha will then create a sort of certificate called a `desc:handoff-give` which includes important information from both node A and node C's session and node A and node B's session, as well as the ID of the gift she left at C. Node B can create a connection to node C if it doesn't already have one and create a certificate called `desc:handoff-receive` which includes the `desc:handoff-give` certificate A give B, this certificate it delivers to C and will get in return the gift if all went successfully.
+
+That's a lot to take in so we can look at each step one by one and see what's actually happening. The first thing to do before that is to look at the message Alice sends to Ben, the `desc:handoff-give` is used in place of gallery object on Node C the message conceptually would look like this:
+
+```
+<op:deliver-only <reference to Ben (desc:import-object)>  [<desc:handoff-give representing the reference to the gallery>]>
+```
+
+#### Depositing the gift
+
+![](./handoffs-3party2.webm)
+
+This is done by the gifter, in our example Alisha's node, Node A. Each node needs a per-session gifting table which stores a gift ID to some reference to an object or promise. The gift ID is just a positive integer (usually incrementing integer) which is unique to the session which represents the reference being shared. When a handoff is initiated it's to a specific object on a node, not just the Node, the gifting mechanism is providing a secure way to share a specific reference to the receiver. The way a node deposits a gift is on the bootstrap object which has a specific `deposit-gift` method.
```

Comment:
Which bootstrap object? The one Alisha got from Node C, the Exporter?

btw the `deposit-gift` method on the bootstrap object is nearly the same as the `provideFor()` on the NonceLocator in CapTP of E.